### PR TITLE
[SYCL] sub-devices selected by ONEAPI_DEVICE_SELECTOR as root devices

### PR DESCRIPTION
**Pull request by cperkinsintel**
All devices available when using ONEAPI_DEVICE_SELECTOR are root devices, even those which are gotten via the sub-device selection choices ( e.g. `ONEAPI_DEVICE_SELECTOR=level_zero:*.*` ).  In this PR we are ensuring that those devices pretend to be root devices. 

Tests for this are here: https://github.com/intel/llvm-test-suite/pull/1346

### CODE REVIEW

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1346

File: sycl/source/device.cpp [hunk 1]
```diff
@@ -142,6 +142,19 @@ device::get_info() const {
...
 
+template <> device device::get_info<info::device::parent_device>() const {
+  // With ONEAPI_DEVICE_SELECTOR the impl.MRootDevice is preset and may be
+  // overridden (ie it may be nullptr on a sub-device) The PI of the sub-devices
+  // have parents, but we don't want to return them. They must pretend to be
+  // parentless root devices.
+  if (impl->isRootDevice())
+    throw invalid_object_error(
+        "No parent for device because it is not a subdevice",
+        PI_ERROR_INVALID_DEVICE);
```
Comment (steffenlarsen) [inline sycl/source/device.cpp:153]: Would it make sense to add a test to preserve this behavior?

Comment (cperkinsintel) [reply sycl/source/device.cpp:153]: You mean like https://github.com/intel/llvm-test-suite/pull/1346 ?

Comment (steffenlarsen) [reply sycl/source/device.cpp:153]: I mean exactly like that! I need stronger glasses it seems.

**Review by steffenlarsen — APPROVED** [review]

**Comment by steffenlarsen** [comment]
@cperkinsintel - post-link is failing on macOS and Linux after merge. Could you please investigate?

**Pull request merged**